src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -139,19 +140,36 @@ export class TerminalInstance implements ITerminalInstance {
 		this._initDimensions();
 		this._createProcess(this._shellLaunchConfig);
 		this._createXterm();
+		if (this._shellLaunchConfig) {
+			this.setTitle(path.basename(this._shellLaunchConfig.executable), true);
+		}
 
 		if (platform.isWindows) {
```

Comment:
But if mac's executable is set to `/bin/bash`, it might be `bash` -> `/bin/bash` (some time in the future)?